lib/test_kfence.c

```diff
@@ -31,11 +31,31 @@ int do_test(size_t size)
 	return 0;
 }
 
+int do_test_uaf(size_t size)
+{
+	int i;
+	volatile char *c;
+	void **buffers;
+
+	buffers = kmalloc_array(MAX_ITER, sizeof(void *), GFP_KERNEL);
+	for (i = 0; i < MAX_ITER; i++) {
+		buffers[i] = kmalloc(size, GFP_KERNEL);
```

Comment:
I don't understand why you need 'buffers[i]'. They do not appear to be reused in later iterations or elsewhere.

Why not just 'char *buffer = kmalloc...' ?

Reply:
Yes, I rewrote this test a bit, and now buffers[i] aren't needed anymore.
Actually, I wanted to introduce a function that checks that a buffer belongs to kfence pool, so that we only test such allocations. Let me try that...